lib/pushmi_pullyu/solr_fetcher.rb

```diff
@@ -0,0 +1,55 @@
+require 'pushmi_pullyu'
+require 'net/http'
+require 'csv'
+
+class PushmiPullyu::SolrFetcher
+
+  attr_accessor :config
+
+  def initialize(config = nil)
+    self.config = config || PushmiPullyu.options
+  end
+
+  def query_url
+    "#{config[:solr][:url]}/select"
+  end
+
+  def run_query(query, fields: nil, csv: false, url_extra: nil)
+    # Return fetched results, else raise an error
+    url = "#{query_url}?q=#{query}"
+    url += "&fl=#{fields}" if fields
+    url += '&wt=csv' if csv
+    url += url_extra if url_extra
+    uri = URI(url)
+
+    request = Net::HTTP::Get.new(uri)
+
+    response = Net::HTTP.start(uri.hostname, uri.port) do |http|
+      http.request(request)
+    end
+
+    return response.body if response.is_a?(Net::HTTPSuccess)
+
+    raise PushmiPullyu::SolrFetchError
+  end
+
+  # Return output as an array of hashes
+  def fetch_query_array(query, fields: nil, url_extra: nil)
+    out = run_query(query, fields: fields, url_extra: url_extra, csv: true)
+    results = []
+    header = nil
+    CSV.parse(out) do |row|
```

Comment:
Meh, just doing what Peter's example script does (it asked for CSV, so I'm asking for CSV).